include/drivers/sensor/sgp40.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2021, Leonard Pollak
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Extended public API for Sensirion's SGP40 gas sensor
+ *
+ * This exposes an API to the on-chip heater which is specific to
+ * the application/environment and cannot be expressed within the
+ * sensor driver abstraction.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_SENSOR_SGP40_H_
+#define ZEPHYR_INCLUDE_DRIVERS_SENSOR_SGP40_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <drivers/sensor.h>
+
+struct sgp40_compensation_data {
+	/* Temperature in °C */
+	int16_t t;
+	/* Relative Humidity in % */
+	int16_t rh;
+};
+
+/**
+ * @brief Updates the compensation data.
+ *
+ * This has to be done before every measurement.
+ *
+ * @param dev - Pointer to the sensor device
+ * @param comp_data - The compensation data
+ *
+ * @return 0 if successful, negative errno code if failure.
+ */
+int sgp40_update_compensation_data(const struct device *dev,
```

Comment:
sensor attributes (using private channels) could be used instead of a custom public API

Reply:
using private channels would involve setting two attributes each time one wants to update the compensation data.
i think it is nicer to have this API for that.